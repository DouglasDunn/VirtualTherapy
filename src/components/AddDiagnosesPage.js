import React from 'react';
import { connect } from 'react-redux';
import DiagnosesForm from './DiagnosesForm';
// import { startAddExpense } from '../actions/expenses';

export class AddDiagnosesPage extends React.Component {
  // onSubmit = (expense) => {
  //   this.props.startAddExpense(expense);
  //   this.props.history.push('/');
  // };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Diagnoses or Symptoms</h1>
          </div>
        </div>
        <div className="content-container">
          <DiagnosesForm />
        </div>
      </div>
    );
  }
}

// <ExpenseForm
//   onSubmit={this.onSubmit}
// />

// const mapDispatchToProps = (dispatch) => ({
//   startAddExpense: (expense) => dispatch(startAddExpense(expense))
// });

export default AddDiagnosesPage;
//export default connect(undefined, mapDispatchToProps)(AddExpensePage);
