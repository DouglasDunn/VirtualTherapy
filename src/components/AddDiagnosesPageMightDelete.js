import React from 'react';
import { connect } from 'react-redux';
import DiagnosesForm from './DiagnosesForm';
import { startAddDiagnoses } from '../actions/diagnoses';
import DiagnosesListItem from './DiagnosesListItem';

export class AddDiagnosesPage extends React.Component {
  onSubmit = (diagnoses) => {
    this.props.startAddDiagnoses(diagnoses);
    this.props.history.push('/diagnoses');
    console.log('add');
    console.log(this.props);
  };
  // nextPage = () => {
  //   this.props.history.push('/diagnoses');
  // };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Diagnoses or Symptoms</h1>
          </div>
        </div>
        <div className="content-container">
          <DiagnosesForm
            onSubmit={this.onSubmit}
          />
        </div>
      </div>
    );
  }
}

// const mapStateToProps = (state) => ({
//   diagnoses: state.diagnoses
// });

const mapDispatchToProps = (dispatch) => ({
  startAddDiagnoses: (diagnoses) => dispatch(startAddDiagnoses(diagnoses))
});

// export default connect(mapStateToProps, mapDispatchToProps)(AddDiagnosesPage);

export default connect(undefined, mapDispatchToProps)(AddDiagnosesPage);

// <div className="content-container">
//   <div className="list-header">
//     <div className="show-for-mobile">Diagnoses</div>
//     <div className="show-for-desktop">Diagnoses</div>
//     <div className="show-for-desktop">Time</div>
//   </div>
//   <div className="list-body">
//     {
//       this.props.diagnoses.length === 0 ? (
//         <div className="list-item list-item--message">
//           <span>No Diagnoses yet, click the button above to add them.</span>
//         </div>
//       ) : (
//         this.props.diagnoses.map((diagnoses) => {
//           return <DiagnosesListItem key={diagnoses.id} {...diagnoses} />;
//         })
//       )
//     }
//   </div>
// </div>
// <div className="content-container">
//   <button onClick={this.nextPage} className="button">Next</button>
// </div>
