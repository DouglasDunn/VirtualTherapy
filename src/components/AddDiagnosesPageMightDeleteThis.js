import React from 'react';
import { connect } from 'react-redux';
import { startAddDiagnoses } from '../actions/diagnoses';

const initialErrorState = {
  diagnosedError: '',
  diagnosesListError: ''
};

export class AddDiagnosesPage extends React.Component {
  constructor(props) {
    super(props);
    let radioButton = 'no';
    let diagnosesArray;
    if (this.props.diagnoses) {
      if (Array.isArray(this.props.diagnoses)) {
        radioButton = 'yes';
        console.log(Array.isArray(this.props.diagnoses));
        console.log(this.props.diagnoses);
        diagnosesArray = this.props.diagnoses.map(({ diagnoses }) => diagnoses).join(', ');
      }
    }

    this.state = {
      diagnosed: this.props.diagnoses ? radioButton: '',
      diagnosesList: this.props.diagnoses ? diagnosesArray : '',
      ...initialErrorState
    };
  }

  onDiagnosedChange = (e) => {
    const diagnosed = e.target.value;
    this.setState(() => ({ diagnosed }));
  };

  onDiagnosesListChange = (e) => {
    const diagnosesList = e.target.value;
    this.setState(() => ({ diagnosesList }));
  };

  validate = () => {
    let diagnosedError = '';
    let diagnosesListError = '';

    if (!this.state.diagnosed) {
      diagnosedError = 'Please choose an option';
    }

    if ((this.state.diagnosed === 'yes') && !this.state.diagnosesList) {
      diagnosesListError = 'diagnoses list cannot be blank';
    } else {
      diagnosesListError = '';
    }

    if (diagnosedError || diagnosesListError) {
      this.setState({ diagnosedError, diagnosesListError });
      return false;
    }

    return true;
  };

  handlePrevious = () => {
    this.props.history.push('/add-drug-allergies');
  };

  onSubmit = (e) => {
    e.preventDefault();
    const isValid = this.validate();
    if (isValid) {
      if (this.state.diagnosed === 'no') {
        this.props.startAddDiagnoses({
          diagnosed: this.state.diagnosed,
          diagnosesList: ''
        });
      } else {
        console.log(this.state.diagnosesList);
        this.props.startAddDiagnoses({
          diagnosed: this.state.diagnosed,
          diagnosesList: this.state.diagnosesList
        });
      }
      this.props.history.push('/add-medication');
    }
  };

  renderDiagnosesList = () => {
    if (this.state.diagnosed === 'yes') {
      return (
        <div>
          {this.state.diagnosesListError && <p className="form__error">{this.state.diagnosesListError}</p>}
          <textarea
            placeholder="Type in all your symptoms and diagnoses separated by a comma"
            className="textarea"
            value={this.state.diagnosesList}
            onChange={this.onDiagnosesListChange}
          >
          </textarea>
        </div>
      );
    }
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Diagnoses or Symptoms</h1>
          </div>
        </div>
        <div className="content-container">
          <form className="form">
            {this.state.diagnosedError && <p className="form__error">{this.state.diagnosedError}</p>}
            Has a doctor ever diagnosed you with a mental health disease?
            <label>
              <input
                type="radio"
                checked={this.state.diagnosed === 'yes'}
                className="radio-input"
                value="yes"
                onChange={this.onDiagnosedChange}
              />
              Yes
            </label>
            <label>
              <input
                type="radio"
                checked={this.state.diagnosed === 'no'}
                className="radio-input"
                value="no"
                onChange={this.onDiagnosedChange}
              />
              No
            </label>
            {this.renderDiagnosesList()}
          </form>
          <div>
            <button onClick={this.handlePrevious} className="button">Previous</button>
            <button onClick={this.onSubmit} className="button">Next</button>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  diagnoses: state.diagnoses
});

const mapDispatchToProps = (dispatch) => ({
  startAddDiagnoses: (diagnoses) => dispatch(startAddDiagnoses(diagnoses))
});

export default connect(mapStateToProps, mapDispatchToProps)(AddDiagnosesPage);
