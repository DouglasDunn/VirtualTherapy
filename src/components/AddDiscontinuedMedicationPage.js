import React from 'react';

export class AddDiscontinuedMedicationPage extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      discontinuedMedication: '' ? 'yes' : 'no'
    };
  }

  onDiscontinuedMedicationChange = (e) => {
    const discontinuedMedication = e.target.value;
    this.setState(() => ({ discontinuedMedication }));
  };

  handlePrevious = () => {
    this.props.history.push('/add-medication');
  };

  onSubmit = (e) => {
    e.preventDefault();

    this.props.history.push('/exercise-questions');
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Discontinued Medications</h1>
          </div>
        </div>
        <div className="content-container">
          <form className="form">
            Have you stopped taking any medications due to side effects?
            <label>
              <input
                type="radio"
                checked={this.state.discontinuedMedication === 'yes'}
                className="radio-input"
                value="yes"
                onChange={this.onDiscontinuedMedicationChange}
              />
              Yes
            </label>
            <label>
              <input
                type="radio"
                checked={this.state.discontinuedMedication === 'no'}
                className="radio-input"
                value="no"
                onChange={this.onDiscontinuedMedicationChange}
              />
              No
            </label>
          </form>
          <div>
            <button onClick={this.handlePrevious} className="button">Previous</button>
            <button onClick={this.onSubmit} className="button">Next</button>
          </div>
        </div>
      </div>
    );
  }
}

export default AddDiscontinuedMedicationPage;
