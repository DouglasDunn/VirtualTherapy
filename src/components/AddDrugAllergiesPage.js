import React from 'react';
import { connect } from 'react-redux';
import DrugAllergyForm from './DrugAllergyForm';
import { startAddDrugAllergy } from '../actions/drugAllergy';

export class AddDrugAllergiesPage extends React.Component {
  onSubmit = (drugAllergy) => {
    this.props.startAddDrugAllergy(drugAllergy);
    this.props.history.push('/add-diagnoses');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Drug Allergies</h1>
          </div>
        </div>
        <div className="content-container">
          <DrugAllergyForm
            onSubmit={this.onSubmit}
          />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddDrugAllergy: (drugAllergy) => dispatch(startAddDrugAllergy(drugAllergy))
});

export default connect(undefined, mapDispatchToProps)(AddDrugAllergiesPage);
