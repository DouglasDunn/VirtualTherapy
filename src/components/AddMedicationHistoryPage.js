import React from 'react';
import { connect } from 'react-redux';
import MedicationHistoryForm from './MedicationHistoryForm';
import { startAddMedicationHistory } from '../actions/medicationHistory';

export class AddMedicationHistoryPage extends React.Component {
  onSubmit = (medicationHistory) => {
    this.props.startAddMedicationHistory(medicationHistory);
    this.props.history.push('/medication-history');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Medication History</h1>
          </div>
        </div>
        <div className="content-container">
          <MedicationHistoryForm
            onSubmit={this.onSubmit}
          />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddMedicationHistory: (medicationHistory) => dispatch(startAddMedicationHistory(medicationHistory))
});

export default connect(undefined, mapDispatchToProps)(AddMedicationHistoryPage);
