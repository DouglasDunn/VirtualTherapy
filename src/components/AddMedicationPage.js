import React from 'react';
import { connect } from 'react-redux';
import DiagnosesListItem from './DiagnosesListItem';

const initialErrorState = {
  diagnosedError: '',
  diagnosesListError: ''
};

export class AddMedicationPage extends React.Component {
  constructor(props) {
    super(props);

    let count = 0;

    this.props.diagnoses.forEach(({ medication }) => { medication && count++; });

    this.state = {
      prescribed: count ? 'yes': 'no'
    };
  }

  onPrescribedChange = (e) => {
    const prescribed = e.target.value;
    this.setState(() => ({ prescribed }));
  };

  handlePrevious = () => {
    this.props.history.push('/add-diagnoses');
  };

  onSubmit = (e) => {
    e.preventDefault();

    this.props.history.push('/add-discontinued-medication');
  };

  renderDianosesData = () => {
    if (this.state.prescribed === 'yes') {
      return (
        <div>
          <div className="list-header">
            <div className="show-for-mobile">Diagnoses</div>
            <div className="show-for-desktop">Diagnoses</div>
          </div>
          <div className="list-body">
            {
              this.props.diagnoses.map(diagnoses =>
                <DiagnosesListItem key={diagnoses.id} {...diagnoses} />
              )
            }
          </div>
        </div>
      );
    }
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Medication</h1>
          </div>
        </div>
        <div className="content-container">
          <form className="form">
            Are you taking any prescribed medication for your symptoms or diagnoses?
            <label>
              <input
                type="radio"
                checked={this.state.prescribed === 'yes'}
                className="radio-input"
                value="yes"
                onChange={this.onPrescribedChange}
              />
              Yes
            </label>
            <label>
              <input
                type="radio"
                checked={this.state.prescribed === 'no'}
                className="radio-input"
                value="no"
                onChange={this.onPrescribedChange}
              />
              No
            </label>
            {this.renderDianosesData()}
          </form>
          <div>
            <button onClick={this.handlePrevious} className="button">Previous</button>
            <button onClick={this.onSubmit} className="button">Next</button>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  diagnoses: state.diagnoses
});
//
// const mapDispatchToProps = (dispatch) => ({
//   startAddDiagnoses: (diagnoses) => dispatch(startAddDiagnoses(diagnoses))
// });
//
export default connect(mapStateToProps)(AddMedicationPage);

//********** Figure this out later on **************************
// renderMedicationForm = () => {
//   if (this.state.prescribed === 'yes') {
//     return this.props.diagnoses.map(({ diagnoses, id }) => (
//       <div key={id}>
//         <p>{diagnoses}</p>
//         <input
//           type="text"
//           id={id}
//           placeholder="Medication"
//           className="text-input"
//           value={this.state.medication}
//           onChange={this.onMedicationChange}
//         />
//         <input
//           type="text"
//           placeholder="Time"
//           className="text-input"
//           value={this.state.time}
//           onChange={this.onTimeChange}
//         />
//       </div>
//     ));
//   }
// };
