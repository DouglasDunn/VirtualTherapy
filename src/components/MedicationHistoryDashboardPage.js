import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import MedicationHistoryListItem from './MedicationHistoryListItem';

export const MedicationHistoryDashboardPage = (props) => (
  <div>
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">Medication History</h1>
        <Link className="button" to="/add-medication-history">
          Add Medication
        </Link>
      </div>
    </div>
    <div className="content-container">
      <div className="list-header">
        <div className="show-for-mobile">Medication</div>
        <div className="show-for-desktop">Medication</div>
        <div className="show-for-desktop">Dose</div>
        <div className="show-for-desktop">Purpose</div>
        <div className="show-for-desktop">Side Effects</div>
        <div className="show-for-desktop">Current? If not, why?</div>
      </div>
      <div className="list-body">
      {
        props.medicationHistories.length === 0 ? (
          <div className="list-item list-item--message">
            <span>No medication history, click the button above to add them.</span>
          </div>
        ) : (
          props.medicationHistories.map((medicationHistory) => {
            return <MedicationHistoryListItem key={medicationHistory.id} {...medicationHistory} />;
          })
        )
      }
      </div>
    </div>
  </div>
);

const mapStateToProps = (state) => {
  return {
    medicationHistories: state.medicationHistories
  };
};

export default connect(mapStateToProps)(MedicationHistoryDashboardPage);
