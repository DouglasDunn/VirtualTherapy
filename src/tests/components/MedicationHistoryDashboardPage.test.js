import React from 'react';
import { shallow } from 'enzyme';
import { MedicationHistoryDashboardPage } from '../../components/MedicationHistoryDashboardPage';
import medicationHistories from '../fixtures/medicationHistories';

test('should render MedicationHistoryDashboardPage with medicationHistories', () => {
  const wrapper = shallow(<MedicationHistoryDashboardPage medicationHistories={medicationHistories} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render MedicationHistoryDashboardPage with empty message', () => {
  const wrapper = shallow(<MedicationHistoryDashboardPage medicationHistories={[]} />);
  expect(wrapper).toMatchSnapshot();
});
