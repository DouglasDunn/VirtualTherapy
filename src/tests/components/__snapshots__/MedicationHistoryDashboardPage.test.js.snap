// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MedicationHistoryDashboardPage with empty message 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
        Medication History
      </h1>
      <Link
        className="button"
        to="/add-medication-history"
      >
        Add Medication
      </Link>
    </div>
  </div>
  <div
    className="content-container"
  >
    <div
      className="list-header"
    >
      <div
        className="show-for-mobile"
      >
        Medication
      </div>
      <div
        className="show-for-desktop"
      >
        Medication
      </div>
      <div
        className="show-for-desktop"
      >
        Dose
      </div>
      <div
        className="show-for-desktop"
      >
        Purpose
      </div>
      <div
        className="show-for-desktop"
      >
        Side Effects
      </div>
      <div
        className="show-for-desktop"
      >
        Current? If not, why?
      </div>
    </div>
    <div
      className="list-body"
    >
      <div
        className="list-item list-item--message"
      >
        <span>
          No medication history, click the button above to add them.
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render MedicationHistoryDashboardPage with medicationHistories 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
        Medication History
      </h1>
      <Link
        className="button"
        to="/add-medication-history"
      >
        Add Medication
      </Link>
    </div>
  </div>
  <div
    className="content-container"
  >
    <div
      className="list-header"
    >
      <div
        className="show-for-mobile"
      >
        Medication
      </div>
      <div
        className="show-for-desktop"
      >
        Medication
      </div>
      <div
        className="show-for-desktop"
      >
        Dose
      </div>
      <div
        className="show-for-desktop"
      >
        Purpose
      </div>
      <div
        className="show-for-desktop"
      >
        Side Effects
      </div>
      <div
        className="show-for-desktop"
      >
        Current? If not, why?
      </div>
    </div>
    <div
      className="list-body"
    >
      <MedicationHistoryListItem
        current="yes"
        dose="10mg"
        id="1"
        key="1"
        medication="zoloft"
        purpose="depression"
        sideEffects="drowsiness, agitation, tiredness, loss of appetite"
      />
      <MedicationHistoryListItem
        current="yes"
        dose="50mg"
        id="2"
        key="2"
        medication="tylenol"
        purpose="headache"
        sideEffects="itchy, diahrea, constipation"
      />
      <MedicationHistoryListItem
        current="no, because I lost too much weight"
        dose="100mg"
        id="3"
        key="3"
        medication="advil"
        purpose="toothache"
        sideEffects="drowsiness, increase in appetite, weight loss"
      />
    </div>
  </div>
</div>
`;
